modules/luci-mod-status/luasrc/controller/admin/status.lua

```diff
@@ -183,3 +184,13 @@ function action_nameinfo(...)
 		limit = 1000
 	}) or { })
 end
+
+function act_kick()
+	local iface = luci.http.formvalue("iface")
+	local mac = luci.http.formvalue("mac")
+	if iface and mac then
+		local cmd = string.format("ubus call hostapd.%s del_client \"{'addr':'%s', 'reason':5, 'deauth':true, 'ban_time':60000}\"", iface, mac)
```

Comment:
This is a textbook example of a root command injection vulnerability. Please escape `iface` and `bssid` through [`luci.util.shellquote()`](http://htmlpreview.github.io/?http://raw.githubusercontent.com/openwrt/luci/master/documentation/api/modules/luci.util.html#shellquote)

Reply:
Actually, there's an even better way. LuCI has a native ubus bindings, so you can replace the entire shell code with this:

```
luci.util.ubus("hostapd.%s" % iface, "del_client", {
  addr = bssid,
  deauth = true,
  reason = 5,
  ban_time = 60000
})
```